Index: library/Zend/Controller/Request/Http.php
===================================================================
--- library/Zend/Controller/Request/Http.php	(Revision 23861)
+++ library/Zend/Controller/Request/Http.php	(Arbeitskopie)
@@ -623,17 +623,26 @@
                 $requestUri = substr($requestUri, 0, $pos);
             }
 
-            $requestUri = urldecode($requestUri);
+            $requestUriDecoded = urldecode($requestUri);
 
             if (null !== $baseUrl
-                && ((!empty($baseUrl) && 0 === strpos($requestUri, $baseUrl))
-                    || empty($baseUrl))
-                    && false === ($pathInfo = substr($requestUri, strlen($baseUrl)))
+                && (  !empty($baseUrl) && 0 === strpos($requestUriDecoded, $baseUrl)  )
             ){
-                // If substr() returns false then PATH_INFO is set to an empty string
-                $pathInfo = '';
-            } elseif (null === $baseUrl
-                    || (!empty($baseUrl) && false === strpos($requestUri, $baseUrl))
+                // try to find out which part is encoded
+                if (0 === strpos($requestUri, $this->_baseUrl)) {
+                    // both parts don't need encoding
+                    $pathInfo = substr($requestUri, strlen($this->_baseUrl));
+                } elseif (0 === strpos($requestUriDecoded, $this->_baseUrl)) {
+                    // request_uri needs encoding for match
+                    $pathInfo = substr($requestUri, strlen(urlencode($this->_baseUrl)));
+                } elseif (0 === strpos($requestUri, $baseUrl)) {
+                    // base_url need encoding for match
+                    $pathInfo = substr($requestUri, strlen(urldecode($this->_baseUrl)));
+                } else {
+                    $pathInfo = '';
+                }
+            } elseif (null === $baseUrl || empty($baseUrl)
+                    || (!empty($baseUrl) && false === strpos($requestUriDecoded, $baseUrl))
             ) {
                 $pathInfo = $requestUri;
             }
Index: library/Zend/Form/Element.php
===================================================================
--- library/Zend/Form/Element.php	(Revision 23861)
+++ library/Zend/Form/Element.php	(Arbeitskopie)
@@ -623,11 +623,6 @@
      */
     public function getLabel()
     {
-        $translator = $this->getTranslator();
-        if (null !== $translator) {
-            return $translator->translate($this->_label);
-        }
-
         return $this->_label;
     }
 
